src/XMakeBuildEngine/BackEnd/Node/NodeConfiguration.cs

```diff
@@ -142,7 +142,7 @@ public void Translate(INodePacketTranslator translator)
             translator.Translate(ref _nodeId);
             translator.Translate(ref _buildParameters, BuildParameters.FactoryForDeserialization);
             translator.TranslateArray(ref _forwardingLoggers, LoggerDescription.FactoryForTranslation);
-#if FEATURE_APPDOMAIN
+#if FEATURE_BINARY_SERIALIZATION
```

Comment:
And you are correct, _appDomainSetup is available only for FEATURE_APPDOMAIN, so, it should be a combination.